js/buttons.js

```diff
@@ -263,5 +271,93 @@
     }, 200);
   };
 
+  var initOverlayBtn = function(btn) {
+    // buildiing overlay
+    var overlay = $('<div class="button-overlay"></div>');
+    overlay.css({
+      'display': 'none',
+      'z-index': 1000,
+      'width': '0px',
+      'height': '0px',
+      'position': 'fixed',
+      'background-color': 'rgba(0, 0, 0, 0.5)',
+      'transition': 'all ease 700ms'
+    });
+    $('body').append(overlay);
+
+    // keeping user z-index if set
+    var defaultZIndex = '';
+
+    // listening for .is-loading class
+    var observer = new MutationObserver(function(mutations) {
+      mutations.forEach(function(mutation) {
+        if (mutation.attributeName === "class") {
+          // is-loading
+          if (btn.hasClass('is-loading')) {
+            console.info('loading activated');
+            btn.addClass('is-loading');
+            defaultZIndex = (btn.css('z-index') ? btn.css('z-index') : '');
+            btn.css('z-index', 1001);
+            // initial overlay position
+            overlay.css({
+              'display': 'block',
+              'border-radius': '5px',
+              'width': parseInt(btn.innerWidth())+'px',
+              'height': parseInt(btn.innerHeight())+'px',
+              'top': (btn.offset().top - $(window).scrollTop()) + 'px',
+              'left': btn.offset().left + 'px'
+            });
+            // when overlay initial position is set, show it
+            // timer is here too avoid css reflow
+            setTimeout(function(){
+              overlay.css({
+                'border-radius': '0px',
+                'width': '100%',
+                'height': '100%',
+                'top': '0px',
+                'left': '0px'
+              });
+            }, 100);
+          // !is-loading
+          } else if (!btn.hasClass('is-loading')) {
+            console.info('loading de-activated');
+            // make overlay return too btn
+            overlay.css({
+              'border-radius': '5px',
+              'width': parseInt(btn.innerWidth())+'px',
+              'height': parseInt(btn.innerHeight())+'px',
+              'top': parseInt(btn.offset().top - $(window).scrollTop()) + 'px',
+              'left': parseInt(btn.offset().left) + 'px'
+            });
+            // once animation is done, reset btn z-index
+            setTimeout(function(){
+              overlay.css('display', 'none');
+              btn.css('z-index', defaultZIndex);
+            }, 700);
+          }
+        }
+      });
+    });
+
+    observer.observe(btn[0],  {
+      attributes: true
+    });
+
+    // if user delete btn with $(btn).remove();
+    btn.on('destroyed', function(){
+      overlay.remove();
+      observer.disconnect();
+      console.info('overlay removed');
+    });
+  };
+
+  // call $(el).on('destroyed') when $(el).remove() is used
```

Comment:
@tomscholz or maybe @Dogfalo 
I need some advice on this

How/Where do you define _"global function"_ like this in Materializecss ? 

Is using `js/init.js` a good idea ?